discord_bot2.py

```diff
@@ -180,9 +278,16 @@ async def on_message(self, message):
         if message.content.startswith('//'):
             return
 
+        if message.channel.name == 'control-duck':
+            await self.control_on_message(message)
+            return
+
         # if the message is in a listen channel, create a thread
         if message.channel.name in self.prompts:
-            prefix = self.prompts[message.channel.name]
+            prefix = ""
+            if message.channel.category.name.lower() in self.prompts:
+                prefix += self.prompts[message.channel.category.name.lower()]
+            prefix += self.prompts[message.channel.name]
```

Comment:
I think this is a clever idea, but not quite how I would recommend doing it. I would take this part out of this PR and do a separate PR to address hierarchical prompts.